Standalone: Panic: Assignment to entry in nil map

It seems to be possible to get Weaviate into a situation where it can't serve any requests anymore, but instead panics with the following stacktrace:

```
2020-11-20 09:54:53.056083 I | http: panic serving 172.28.0.1:57664: assignment to entry in nil map
goroutine 678 [running]:
net/http.(*conn).serve.func1(0xc000d9a1e0)
        /usr/local/go/src/net/http/server.go:1800 +0x139
panic(0x1264760, 0x1756b00)
        /usr/local/go/src/runtime/panic.go:975 +0x3e3
github.com/semi-technologies/weaviate/adapters/repos/db/storobj.(*Object).SearchResult(0xc000902dc0, 0x0, 0x0, 0x0, 0x0, 0x2159dd8)
        /go/src/github.com/semi-technologies/weaviate/adapters/repos/db/storobj/storage_object.go:202 +0xee
github.com/semi-technologies/weaviate/adapters/repos/db.(*DB).objectByID(0xc000aa8900, 0x1791be0, 0xc00089f830, 0x140663d, 0x5, 0xc000858c30, 0x24, 0x2159dd8, 0x0, 0x0, ...)
        /go/src/github.com/semi-technologies/weaviate/adapters/repos/db/crud.go:153 +0x1c2
github.com/semi-technologies/weaviate/adapters/repos/db.(*DB).ThingByID(0xc000aa8900, 0x1791be0, 0xc00089f830, 0xc000858c30, 0x24, 0x2159dd8, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/semi-technologies/weaviate/adapters/repos/db/crud.go:87 +0xca
github.com/semi-technologies/weaviate/usecases/kinds.(*Manager).getThingFromRepo(0xc0007222c0, 0x1791be0, 0xc00089f830, 0xc000858c30, 0x24, 0x0, 0x0, 0x0, 0x0, 0x14057b7, ...)
        /go/src/github.com/semi-technologies/weaviate/usecases/kinds/get.go:114 +0xab
github.com/semi-technologies/weaviate/usecases/kinds.(*Manager).GetThing(0xc0007222c0, 0x1791be0, 0xc00089f830, 0x0, 0xc000858c30, 0x24, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/semi-technologies/weaviate/usecases/kinds/get.go:44 +0x2a5
github.com/semi-technologies/weaviate/adapters/handlers/rest.(*kindHandlers).getThing(0xc00009a300, 0xc00096f000, 0xc000858c30, 0x24, 0x0, 0x0, 0x0, 0x7f026337a108, 0x1791be0)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/handlers_kinds.go:169 +0x109
github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/things.ThingsGetHandlerFunc.Handle(0xc000214870, 0xc00096f000, 0xc000858c30, 0x24, 0x0, 0x0, 0x0, 0x0, 0x11db5a0)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/things/things_get.go:32 +0x51
github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/things.(*ThingsGet).ServeHTTP(0xc00039d900, 0x178d2e0, 0xc0008cba40, 0xc00096f000)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/things/things_get.go:82 +0x342
github.com/go-openapi/runtime/middleware.NewOperationExecutor.func1(0x178d2e0, 0xc0008cba40, 0xc00096ef00)
        /go/pkg/mod/github.com/go-openapi/runtime@v0.19.24/middleware/operation.go:28 +0x75
net/http.HandlerFunc.ServeHTTP(0xc0003250e0, 0x178d2e0, 0xc0008cba40, 0xc00096ef00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest.makeSetupMiddlewares.func1.1(0x178d2e0, 0xc0008cba40, 0xc00096ef00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/middlewares.go:35 +0xb3
net/http.HandlerFunc.ServeHTTP(0xc00039daa0, 0x178d2e0, 0xc0008cba40, 0xc00096ef00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/go-openapi/runtime/middleware.NewRouter.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/pkg/mod/github.com/go-openapi/runtime@v0.19.24/middleware/router.go:77 +0x356
net/http.HandlerFunc.ServeHTTP(0xc0004dcbe0, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/go-openapi/runtime/middleware.Redoc.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/pkg/mod/github.com/go-openapi/runtime@v0.19.24/middleware/redoc.go:72 +0x286
net/http.HandlerFunc.ServeHTTP(0xc000e52d40, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/go-openapi/runtime/middleware.Spec.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/pkg/mod/github.com/go-openapi/runtime@v0.19.24/middleware/spec.go:46 +0x188
net/http.HandlerFunc.ServeHTTP(0xc000e52d80, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rs/cors.(*Cors).Handler.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/pkg/mod/github.com/rs/cors@v1.5.0/cors.go:200 +0x1af
net/http.HandlerFunc.ServeHTTP(0xc0004dd0e0, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest/swagger_middleware.AddMiddleware.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/swagger_middleware/swagger_middleware.go:37 +0xc9
net/http.HandlerFunc.ServeHTTP(0xc000ae3d70, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest.makeAddLogging.func1.1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/middlewares.go:82 +0x2e9
net/http.HandlerFunc.ServeHTTP(0xc000ae3da0, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest.addPreflight.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/middlewares.go:96 +0x71
net/http.HandlerFunc.ServeHTTP(0xc0004dd140, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest.addLiveAndReadyness.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/middlewares.go:112 +0xa7
net/http.HandlerFunc.ServeHTTP(0xc0004dd160, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/semi-technologies/weaviate/adapters/handlers/rest.addHandleRoot.func1(0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /go/src/github.com/semi-technologies/weaviate/adapters/handlers/rest/middlewares.go:50 +0x8f
net/http.HandlerFunc.ServeHTTP(0xc0004dd1a0, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2041 +0x44
net/http.serverHandler.ServeHTTP(0xc0007596c0, 0x178d2e0, 0xc0008cba40, 0xc00096ed00)
        /usr/local/go/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc000d9a1e0, 0x1791b20, 0xc00064f4c0)
        /usr/local/go/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2962 +0x35c
```

We are not fully aware yet how to reproduce this bug, but it seems it also occurs on our regular Journey Test Suite when running it with Standalone, in particular, this is the test that fails:

```
ok      github.com/semi-technologies/weaviate/genesis/test/acceptance   0.250s [no tests to run]
before first
--- FAIL: Test_Actions (6.81s)
    --- FAIL: Test_Actions/updating_actions (6.43s)
        --- FAIL: Test_Actions/updating_actions/can_patch_action_with_cref (6.35s)
            eventually_equal.go:56: waiting for 4s, but never succeeded:


                        Error Trace:    eventually_equal.go:46
                                                                helper.go:102
                                                                update_test.go:113
                        Error:          Not equal:
                                        expected: true
                                        actual  : false
            assertions.go:26: Failed to perform request! Error: *Error {
                 "Op": "Get",
                 "URL": "http://:8080/v1/things/eb7ea7a8-5182-4fa3-83be-48df010fd7b7",
                 "Err": {}
                } (Original error Get "http://:8080/v1/things/eb7ea7a8-5182-4fa3-83be-48df010fd7b7": EOF). Response: null

```

--- a/adapters/repos/db/shard_write_merge.go
+++ b/adapters/repos/db/shard_write_merge.go
@@ -113,7 +113,7 @@ func mergeProps(previous *storobj.Object,
 	merge kinds.MergeDocument) *storobj.Object {
 	next := *previous
 	schema, ok := next.Schema().(map[string]interface{})
-	if !ok {
+	if !ok || schema == nil {
 		schema = map[string]interface{}{}
 	}
 

--- a/adapters/repos/db/storobj/storage_object.go
+++ b/adapters/repos/db/storobj/storage_object.go
@@ -195,11 +195,12 @@ func (ko *Object) VectorWeights() models.VectorWeights {
 }
 
 func (ko *Object) SearchResult(underscore traverser.UnderscoreProperties) *search.Result {
-	schema := ko.Schema()
-	if schema == nil {
-		schema = map[string]interface{}{}
+	schemaMap, ok := ko.Schema().(map[string]interface{})
+	if !ok || schemaMap == nil {
+		schemaMap = map[string]interface{}{}
 	}
-	schema.(map[string]interface{})["uuid"] = ko.ID()
+	schemaMap["uuid"] = ko.ID()
+	ko.SetSchema(schemaMap)
 
 	underscoreProperties := &models.UnderscoreProperties{}
 	if ko.UnderscoreProperties() != nil {

